### PR TITLE
Integrate the gopath hacks

### LINKED ISSUE #435: Use GOPATH hacks to codegen anywhere.
Please see https://github.com/knative/eventing/pull/4621 or https://github.com/knative/serving/pull/10245 for an example.

### PR DESCRIPTION
**Pull request by arghya88**
Signed-off-by: Arghya Sadhu <arghya88@gmail.com>

# Changes

- Use GOPATH hacks to codegen anywhere.

/kind <kind>

Fixes #435

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -18,40 +18,44 @@ set -o errexit
...
+
 # If we run with -mod=vendor here, then generate-groups.sh looks for vendor files in the wrong place.
 export GOFLAGS=-mod=
 
-if [ -z "${GOPATH:-}" ]; then
-  export GOPATH=$(go env GOPATH)
-fi
+echo "=== Update Codegen for $MODULE_NAME"
+
+CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${REPO_ROOT_DIR}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
```
Comment (n3wscott) [inline hack/update-codegen.sh:28]: you can drop this line

File: hack/update-codegen.sh [hunk 2]
```diff
@@ -18,40 +18,44 @@ set -o errexit
...
 
-REPO_ROOT=$(dirname ${BASH_SOURCE})/..
-CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${REPO_ROOT}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
+KNATIVE_CODEGEN_PKG=${KNATIVE_CODEGEN_PKG:-$(cd ${REPO_ROOT_DIR}; ls -d -1 ./vendor/knative.dev/pkg 2>/dev/null || echo ../pkg)}
```
Comment (n3wscott) [inline hack/update-codegen.sh:30]: and this line can be dropped

**Comment by arghya88** [comment]
` Error: SA1019: pkgTest.MatchesAllOf is deprecated: Use the spoof package version  (staticcheck)`

This error seems to be happening for last couple of PRs

**Review by nak3 — APPROVED** [review]
LGTM.

The lint issue was fixed by https://github.com/knative-sandbox/net-istio/pull/442.

**Comment by arghya88** [comment]
@n3wscott let's get this in.

**Pull request merged**